Diplom_2 pull Request

mober2000 opened the pull request.

mober2000 committed:
Создание для пул реквеста

mober2000 committed:
Проверка пул реквеста

mober2000 committed:
Добавил тесты на логирование и сейчас пока разбираюсь с тестами на внесение изменений, почему-то я могу вносить изменения будучи неавторизованным, норма это или нет я не знаю, задать вопрос после празников

mober2000 committed:
Добавил тесты на изменение данных пользователя (авторизованного и неавторизованного)

mober2000 committed:
Коммит с заготовкой на создание заказа

mober2000 committed:
Добавилметод на получение обновленного токена, добавил тесты на создание заказа

mober2000 committed:
Доработал методы на создание заказа

mober2000 committed:
Добавил все тесты указаные в задании, осталось только сделать рефакторинг и облагородить их

mober2000 committed:
Провожу рефакторинг кода автотестов, закончил примерно процентов на 30

mober2000 committed:
Апи методы вынес в отдельный класс, распределил файлы по package, видоизменил все тесты с юзером

mober2000 committed:
Логика тестов готова, проведен рефакторинг, осталось заняться кодстайлингом

mober2000 committed:
Завершил кодстайлинг и добавил анатации аллюр ко всем тестам

mober2000 committed:
Провел кодстайлинг по pojo и api

mober2000 committed:
Проведен кодстайлинг и рефакторинг во всем проекте

mober2000 committed:
Добавил аллюр отчет к тестам
Проект готов, первая попытка сдачи проекта

mober2000 committed:
Небольшие грамматические ошибки
Проект готов, первая попытка сдачи проекта

pomegranate282922 commented on src/main/java/api/Api.java:
```diff
@@ -0,0 +1,90 @@
+        return reqSpec
+                .body(createUserData)
+                .when()
+                .post("auth/register")
```
Урлы и их части выносим как константы.  Так их проще переиспользовать и изменять. Нужно исправить.

pomegranate282922 commented on src/main/java/api/Api.java:
```diff
@@ -0,0 +1,90 @@
+
+    @Step("Запрос на удаление пользователя без ключа авторизации")
+    public ValidatableResponse deleteUserNotKeyRequest() {
+        return reqSpec
```
Отлично! Использование req spec здорово уменьшает дублирование кода.

pomegranate282922 commented on src/main/java/api/Api.java:
```diff
@@ -0,0 +1,90 @@
+                .patch("auth/user")
+                .then();
+    }
+
```
Отлично, что указаны названия для степов.

pomegranate282922 commented on src/main/java/pojo/ingridientdata/IngredientData.java:
```diff
@@ -0,0 +1,24 @@
+package pojo.ingridientdata;
```
Отлично, что выделены  пакеты!

pomegranate282922 commented on src/main/java/universalclasses/RandomGenerator.java:
```diff
@@ -0,0 +1,33 @@
+public class RandomGenerator {
+    private final String emailYandex = setRandomEmail();
+    private final String password = setRandomPassword();
+    private final String name = setRandomName();
```
Отлично, что данные рандомизированы. Это делает тесты независимыми от состояния бд.

pomegranate282922 commented on src/test/java/usertests/CreateUserTests.java:
```diff
@@ -0,0 +1,42 @@
+    @Description("Проверяем не создается ли пользователь c пустыми полями почты, пароля и имени")
+    public void createUserNullFields() {
+        createUser.createUserNullFields(mail, password, name);
+    }
```
Здесь тоже может создасться юзер, если тест словит баг. И его нужно удалить в таком случае.

pomegranate282922 commented on src/test/java/usertests/LoginUserTests.java:
```diff
@@ -0,0 +1,34 @@
+
+public class LoginUserTests {
+    LoginUser loginUser = new LoginUser();
+    String mail = "testtesttest1234@gmail.com";
```
Нет, так нехорошо. Это делает тесты зависимыми от бд. Нужно создать рандомного юзера перед тестом и удалить после.

pomegranate282922 reviewed (commented):
Все хорошо, но пара замечаний имеется. Прошу их исправить.

mober2000 committed:
Сдача проекта - попытка номер 2
Вынес части юрл в константы
В тесты авторизации добавил независимые от базы методы на создание
пользователя
В тест с созданием пользователя и ошибками, добавил возможность удалять данные если будет баг

pomegranate282922 reviewed (approved):
Отлично, зачёт!

mober2000 committed:
Саня смотри
